sdk/azidentity/doc.go

```diff
@@ -0,0 +1,89 @@
+// +build go1.13
+
+// Copyright 2017 Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by an MIT
+// license that can be found in the LICENSE file.
+
+/*
+Package azidentity implements a set of credential types for use with
+Azure SDK clients that support Azure Active Directory (AAD) token authentication.
+
+The following credential types are included in this module:
+	- AuthorizationCodeCredential
+	- AzureCLICredential
+	- ChainedTokenCredential
+	- ClientCertificateCredential
+	- ClientSecretCredential
+	- DefaultAzureCredential
+	- DeviceCodeCredential
+	- EnvironmentCredential
+	- InteractiveBrowserCredential
+	- ManagedIdentityCredential
+	- UsernamePasswordCredential
+
+By default, the recommendation is that users call NewDefaultAzureCredential() which will
+provide a default ChainedTokenCredential configuration conformed of:
+	- EnvironmentCredential
+	- ManagedIdentityCredential
+	- AzureCLICredential
+Configuration options can be used to exclude any of the previous credentials from the
+DefaultAzureCredential implementation.
+
+Example call to NewDefaultAzureCredential():
+	cred, err := NewDefaultAzureCredential(nil) // pass in nil to get default behavior for this credential
+	if err != nil {
+		// process error
+	}
+	client := armresources.NewResourcesClient(armcore.NewDefaultConnection(cred, nil), "<subscription ID>")
+
+Example call to NewDefaultAzureCredential() with options set:
+	// these options will make sure the AzureCLICredential will not be added to the credential chain
+	opts := DefaultAzureCredentialOptions{ExcludeAzureCLICredential: true}
+	cred, err := NewDefaultAzureCredential(&opts)
+	if err != nil {
+		// process error
+	}
+	client := armresources.NewResourcesClient(armcore.NewDefaultConnection(cred, nil), "<subscription ID>")
+
+Additional configuration of each credential can be done through each credential's
+options type. These options can also be used to modify the default pipeline for each
+credential.
+
+Example pattern for modifying credential options:
+	opts := DefaultClientCertificateCredentialOptions()
+	opts.Password = "<certificate password>"
+	cred, err := azidentity.NewClientCertificateCredential("<tenant ID>", "<client ID>", "<certificate path>", &opts)
+	if err != nil {
+		// process error
+	}
+
+CREDENTIAL AUTHORITY HOSTS
+
+The default authority host for all credentials, except for ManagedIdentityCredential, is the
+AzurePublicCloud host. This value can be changed through the credential options or by specifying
+a different value in an environment variable called AZURE_AUTHORITY_HOST.
```

Comment:
As I recall, there's an ordering issue here where if the value is explicitly set in the code the env var will never be honored.  Would be good to explain that here.

Reply:
Done